connectors/sources/zoom.py

```diff
@@ -0,0 +1,539 @@
+                date_to=format_recording_date(date=end_date),
+                page_size=MEETING_PAGE_SIZE,
+            )
+            async for recordings in self.api_client.scroll(url=url):
+                for recording in recordings.get("meetings", []) or []:
+                    yield recording
+            end_date = start_date
+
+    async def get_channels(self, user_id):
+        url = APIS["CHANNEL"].format(
+            base_url=BASE_URL, user_id=user_id, page_size=CHAT_PAGE_SIZE
+        )
+        async for channels in self.api_client.scroll(url=url):
+            for channel in channels.get("channels", []) or []:
+                yield channel
+
+    async def get_chats(self, user_id, chat_type):
+        # Zoom chat does not retrieve data that is more than 6 months old.
+        end_date = datetime.utcnow()
+        start_date = end_date + timedelta(days=-180)
+        url = APIS["CHAT"].format(
+            base_url=BASE_URL,
+            user_id=user_id,
+            chat_type=chat_type,
+            date_from=format_chat_date(date=start_date),
+            date_to=format_chat_date(date=end_date),
+            page_size=CHAT_PAGE_SIZE,
+        )
+        async for chats in self.api_client.scroll(url=url):
+            for chat in chats.get("messages", []) or []:
+                yield chat
+
+    async def get_file_content(self, download_url):
+        return await self.api_client.content(url=download_url)
+
+
+class ZoomDataSource(BaseDataSource):
+    name = "Zoom"
+    service_type = "zoom"
+
+    def __init__(self, configuration):
+        super().__init__(configuration=configuration)
+        self.configuration = configuration
+
+    def _set_internal_logger(self):
+        self.client.set_logger(self._logger)
+
+    @cached_property
+    def client(self):
+        return ZoomClient(configuration=self.configuration)
+
+    @classmethod
+    def get_default_configuration(cls):
+        return {
+            "account_id": {
+                "label": "Account ID",
+                "order": 1,
+                "type": "str",
+            },
+            "client_id": {
+                "label": "Client ID",
+                "order": 2,
+                "type": "str",
+            },
+            "client_secret": {
+                "label": "Client secret",
+                "order": 3,
+                "sensitive": True,
+                "type": "str",
+            },
+            "fetch_past_meeting_details": {
+                "display": "toggle",
+                "label": "Fetch past meeting details",
+                "order": 4,
+                "tooltip": "Enable this option to fetch past past meeting details. This setting can increase sync time.",
+                "type": "bool",
+                "value": False,
+            },
+        }
+
+    async def validate_config(self):
+        await super().validate_config()
+        await self.client.api_token.get()
```

Comment:
Question: do you call this here to check that the credentials are valid? Or are you actually fetching the token for an upcoming sync job?

If the for validation, a token is already being fetched in `ping`, and `ping` is run alongside `validate_config` ([see here](https://github.com/elastic/connectors-python/blob/ff8ff703c9275c76616a0963ed9ba57a490ab42c/connectors/sync_job_runner.py#L124-L129)) so it's not necessary to check the token again.
If it's for an upcoming sync job, it feels awkward to run during validation.

In either case, I don't think it's necessary to have here.

Reply:
Yes, self.client.api_token.get() kept here as it will create and validate all config params if any value is invalid it will throw an exception.

> If the for validation, a token is already being fetched in ping, and ping is run alongside validate_config ([see here](https://github.com/elastic/connectors-python/blob/ff8ff703c9275c76616a0963ed9ba57a490ab42c/connectors/sync_job_runner.py#L124-L129)) so it's not necessary to check the token again.
Shouldn't configured values remotely validated before sync job or pinging?

Reply:
I can see now that the token is being cached so I'm not as concerned about the duplicate calls.